mpact_bot.py

```diff
@@ -6,37 +6,87 @@
 django.setup()
 
 from telethon import TelegramClient, events
+from telethon.tl import types
 
 from constants import BOT_TOKEN, TELEGRAM_API_HASH, TELEGRAM_API_ID
 from logger import logger
-from mpact.models import ChatData
-from mpact.serializers import ChatDataSerializer
+from mpact.models import Bot, Chat, Individual
+from mpact.serializers import ChatSerializer
 from utils import get_or_none
 
-bot = TelegramClient("bot", TELEGRAM_API_ID, TELEGRAM_API_HASH).start(
+bot_client = TelegramClient("bot", TELEGRAM_API_ID, TELEGRAM_API_HASH).start(
     bot_token=BOT_TOKEN
 )
 
 
-@bot.on(events.ChatAction)
-async def handler(event):
+@bot_client.on(events.NewMessage)
+async def msg_handler(event):
+    """
+    Event handler for new message
+    """
+    try:
+        if event.message.message == "/start":
+            # check if the message is from individual(PeerUser) or group(PeerChat) chat
+            if isinstance(event.message.peer_id, types.PeerUser):
+                current_bot = await bot_client.get_me()
+                user_details = await bot_client.get_entity(
+                    event.message.peer_id.user_id
+                )
+                bots = Bot.objects.filter(bot_id=current_bot.id)
+
+                for bot in bots:
+                    group_id = bot.chat.chat_id
+                    # fetching all the users of a group
+                    users = await bot_client.get_participants(group_id)
+                    for user in users:
+                        # checking if the user is part of any group
+                        if user.first_name == user_details.first_name:
+                            Individual.objects.get_or_create(
+                                bot=bot,
+                                individual_id=user_details.id,
+                                username=user_details.username,
+                                first_name=user_details.first_name,
+                                last_name=user_details.last_name,
+                            )
+                await event.reply("Welcome! How may i help you?")
```

Comment:
I think we should move all messaging to users to a separate file, that would make it easy for the customer to configure them. For example,
```suggestion
                if messages.WELCOME:
                    await event.reply(WELCOME)
```

Reply:
Yes, This is will be better. I will handle this.